src/file/write-file.js

```diff
@@ -1,35 +1,60 @@
+
 var fs = require('fs');
 var Promise = require('bluebird');
 var sourceMap = require('./source-map');
+var convert = require('convert-source-map');
+var path = require('path');
 
 function write(code, map, fileType, outputPath, siteRoot) {
 
-    return new Promise(function(resolve, reject) {
+		map = sourceMap.clean(map, siteRoot);
+		
+        if (map) {

```

Comment:
this file is most of the real change: instead of putting it in the comment we write a separate file if there is a map